### PR TITLE
add blocks cache control

### PR DESCRIPTION
**Pull request by itkg-bfouche**

### CODE REVIEW

File: en/developer_guide/multi_device.rst [hunk 1]
```diff
@@ -6,7 +6,17 @@ Prerequisite
...
-You will also have to set up the device detection, see related doc for an example.
+You will also have to set up the device detection.
+
+OpenOrchestra manages the multi-device with Varnish by adding in the request header a parameter x-ua-device.
+
+In the file `openorchestra.vcl `, Varnish tests User-Agent to send the corresponding tag x-ua-device:
+
+.. code-block:: varnish4
+
+    if (req.http.User-Agent ~ "(?i)android") {
```
Comment (nicolasThal) [inline en/developer_guide/multi_device.rst]: wurfl does this check, those line have only a test purpose

**Comment by itkg-bfouche** [comment]
@nicolasThal

File: en/developer_guide/multi_device.rst [hunk 2]
```diff
@@ -86,3 +86,18 @@ Then, you declare the engine service based on the new class and set the ``templa
...
             alias: templating
+
+Device detection
+----------------
+
+`Wurfl librairy`_ test the User-Agent and normalize devices, the server entry point (webserver or reverse-proxy) must add a parameters ``x-ua-device`` to the request header with the device.
+
+For example if you have a configuration with different template between phone and tablet, the parameters ``x-ua-device`` need to be equals to phone or tablet to display the good template.
+
+OpenOrchestra tests the x-ua-device parameter gotten from the request header, in the FrontBundle trait which overrides the ``render()`` method.
```
Comment (juchi) [inline en/developer_guide/multi_device.rst]: Open Orchestra
x-ua-device should be greyed as above
I personnaly prefer to refer to a method in this form : `ClassName::method()` 
Line to split

File: en/developer_guide/multi_device.rst [hunk 3]
```diff
@@ -86,3 +86,18 @@ Then, you declare the engine service based on the new class and set the ``templa
                 - @request_stack
                 - %open_orchestra_front.devices%
             alias: templating
+
+Device detection
+----------------
+
+`Wurfl librairy`_ test the User-Agent and normalize devices, the server entry point (webserver or reverse-proxy) must add a parameters ``x-ua-device`` to the request header with the device.
+
+For example if you have a configuration with different template between phone and tablet, the parameters ``x-ua-device`` need to be equals to phone or tablet to display the good template.
```
Comment (juchi) [inline en/developer_guide/multi_device.rst]: the line should be split
"templates"
"parameter"
"needs to be equal"
good template => correct template

File: en/developer_guide/multi_device.rst [hunk 4]
```diff
@@ -86,3 +86,18 @@ Then, you declare the engine service based on the new class and set the ``templa
                 - @request_stack
                 - %open_orchestra_front.devices%
             alias: templating
+
+Device detection
+----------------
+
+`Wurfl librairy`_ test the User-Agent and normalize devices, the server entry point (webserver or reverse-proxy) must add a parameters ``x-ua-device`` to the request header with the device.
```
Comment (juchi) [inline en/developer_guide/multi_device.rst]: line to be split

File: en/developer_guide/multi_device.rst [hunk 5]
```diff
@@ -86,3 +86,18 @@ Then, you declare the engine service based on the new class and set the ``templa
...
+
+For example if you have a configuration with different template between phone and tablet, the parameters ``x-ua-device`` need to be equals to phone or tablet to display the good template.
+
+OpenOrchestra tests the x-ua-device parameter gotten from the request header, in the FrontBundle trait which overrides the ``render()`` method.
+If a template is found for this device name, so call ``render()`` method for this template, else check if the parent device template exist.
```
Comment (juchi) [inline en/developer_guide/multi_device.rst]: Line to split

**Comment by itkg-bfouche** [comment]
@nicolasThal @juchi @itkg-ngilain @itkg-nanne

File: en/developer_guide/block_creation.rst [hunk 6]
```diff
@@ -70,3 +70,13 @@ The usage is pretty straightforward :
         --backoffice-icon-dir="src/YourVendor/AcmeBundle/Path/To/Icon/Strategies"
         --backoffice-display-dir="src/YourVendor/AcmeBundle/Path/To/Back/Display/Strategies"
         --no-interaction
+
+Cache control
+-------------
+
+Open Orchestra leverages the power of ESI blocks to optimize page rendering and increase performance.
+In the FrontOffice application, each block is rendered in an ESI block using the render_esi() twig function.
```
Comment (ngilain) [inline en/developer_guide/block_creation.rst]: Front Office

File: en/developer_guide/multi_device.rst [hunk 7]
```diff
@@ -6,7 +6,7 @@ Prerequisite
 
 To enable the multi-device rendering, we assume that the server entry point (webserver or reverse-proxy)
  implements the Wurfl librairy and is able to add a request header with the device type.
```
Comment (ngilain) [inline en/developer_guide/multi_device.rst]: Is an implementation of wurfl trully required ? Open Orchestra only needs to get a request tagged with xua-device, and a twig (or other language) template matching that tag.
With this point of view, wurfl is not a prerequisite, only something to tag the client request.

**Comment by itkg-bfouche** [comment]
@itkg-ngilain

File: en/developer_guide/multi_device.rst [hunk 8]
```diff
@@ -86,3 +86,22 @@ Then, you declare the engine service based on the new class and set the ``templa
...
+Device detection
+----------------
+
+`Wurfl librairy`_ test the User-Agent and normalize devices,
+the server entry point (webserver or reverse-proxy) must add a parameters ``x-ua-device`` to the request header with the device.
+
+For example if you have a configuration with different templates between phone and tablet,
+the parameter ``x-ua-device`` needs to be equal to phone or tablet to display the correct template.
+
+Open Orchestra tests the ``x-ua-device`` parameter gotten from the request header,
```
Comment (ngilain) [inline en/developer_guide/multi_device.rst]: parameter given by the request header
or maybe "got from" but its sounds weird to me

File: en/developer_guide/multi_device.rst [hunk 9]
```diff
@@ -86,3 +86,22 @@ Then, you declare the engine service based on the new class and set the ``templa
...
+in the FrontBundle trait which overrides the ``TwigEngine::render()`` method.
```
Comment (ngilain) [inline en/developer_guide/multi_device.rst]: FrontBundle trait overriding the

**Comment by ngilain** [comment]
What about wurfl implementation requirement ? Is it a hard requirement ? Or is something tagging the response required ?

**Comment by itkg-bfouche** [comment]
@itkg-ngilain Wurfl is a way to make multi-device, it is not a requirement

**Comment by ngilain** [comment]
Ok so maybe this sentence should be rephrased : "To enable the multi-device rendering, we assume that the server entry point (webserver or reverse-proxy) implements the Wurfl librairy and is able to add a request header with the device type."

File: en/developer_guide/multi_device.rst [hunk 10]
```diff
@@ -86,3 +86,22 @@ Then, you declare the engine service based on the new class and set the ``templa
                 - @request_stack
                 - %open_orchestra_front.devices%
             alias: templating
+
+Device detection
+----------------
+
+`Wurfl librairy`_ test the User-Agent and normalize devices,
```
Comment (juchi) [inline en/developer_guide/multi_device.rst]: tests

File: en/developer_guide/multi_device.rst [hunk 11]
```diff
@@ -86,3 +86,22 @@ Then, you declare the engine service based on the new class and set the ``templa
...
+the server entry point (webserver or reverse-proxy) must add a parameters ``x-ua-device`` to the request header with the device.
```
Comment (juchi) [inline en/developer_guide/multi_device.rst]: a parameter
i'd rephrase the end : "to the header of the device's request" (if that's what you meant)

File: en/developer_guide/multi_device.rst [hunk 12]
```diff
@@ -86,3 +86,22 @@ Then, you declare the engine service based on the new class and set the ``templa
...
+
+For example if you have a configuration with different templates between phone and tablet,
```
Comment (juchi) [inline en/developer_guide/multi_device.rst]: For instance

File: en/developer_guide/multi_device.rst [hunk 13]
```diff
@@ -86,3 +86,22 @@ Then, you declare the engine service based on the new class and set the ``templa
...
+the parameter ``x-ua-device`` needs to be equal to phone or tablet to display the correct template.
+
+Open Orchestra tests the ``x-ua-device`` parameter given by the request header,
+in the FrontBundle trait overriding the ``TwigEngine::render()`` method.
+If a template is found for this device name, so call ``TwigEngine::render()`` method for this template,
+else check if the parent device template exist.
+
+For example see ``OpenOrchestra\FrontBundle\Twig\Renderable``.
```
Comment (juchi) [inline en/developer_guide/multi_device.rst]: "For an example" would be more suited.

**Comment by ngilain** [comment]
@juchi , @nicolasThal , @itkg-nanne , @itkg-rboessel , @itkg-alavieille : doc rewrote

File: en/developer_guide/multi_device.rst [hunk 14]
```diff
@@ -1,24 +1,31 @@
 Multi-device
 ============
 
+Open Orchestra is compatible with the adaptive way of rendering web pages. The idea is to serve
+different html versions of the same document according to the device displaying it. For instance
+an iphone version could show less blocks than a comptuter version.
+
+The main steps of the process are the following:
```
Comment (nicolasThal) [inline en/developer_guide/multi_device.rst]: there is a format error, I think that you need to let an empty line before the listing

File: en/developer_guide/multi_device.rst [hunk 15]
```diff
@@ -1,24 +1,31 @@
...
+- Response generation: The page is rendered and served to the client
+
+
 Prerequisite
 ------------
 
-To enable the multi-device rendering, we assume that the server entry point (webserver or reverse-proxy)
- implements the Wurfl librairy and is able to add a request header with the device type.
-You will also have to set up the device detection, see related doc for an example.
+To enable the multi-device rendering, the server entry point (webserver or reverse-proxy) must implement
```
Comment (nicolasThal) [inline en/developer_guide/multi_device.rst]: wurfl is not required, only a library that can detect a device (like Wurfl)

**Comment by ngilain** [comment]
@nicolasThal : update done

**Comment by nicolasThal** [comment]
:+1:
